src/DayPicker.tsx

```diff
@@ -256,6 +263,118 @@ export function DayPicker(props: DayPickerProps) {
 
   const dataAttributes = getDataAttributes(props);
 
+  const rootRef = useRef<HTMLDivElement>(null);
+  const rootElementSnapshot = useRef<HTMLElement>(null);
+  const previousMonths = useRef(months);
+
+  useLayoutEffect(() => {
+    if (months.length === 0) {
+      return;
+    }
+
+    const isSameMonth = dateLib.isSameMonth(
+      months[0].date,
+      previousMonths.current[0].date
+    );
+    const slideLeft = dateLib.isAfter(
+      months[0].date,
+      previousMonths.current[0].date
+    );
+    previousMonths.current = months;
+    if (rootRef.current) {
+      const previousMonthElement =
+        rootElementSnapshot.current?.querySelector(`[data-month]`);
```

Comment:
For supporting multiple months, consider adding a value , such as `data-month="yyyy-mm"`.

Reply:
Done, animation works for multiple months